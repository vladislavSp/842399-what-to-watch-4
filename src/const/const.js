export const MAX_GENRES_NUM = 8;
export const HUNDRED_PERCENT = 100;
export const AVATAR_BASE_PATH = `https://4.react.pages.academy`;
export const REGEX_MAIL = /\S+@\S+\.\S+/;
export const MIN_REVIEW_LENGTH = 50;
export const MAX_REVIEW_LENGTH = 400;

export const GenreFilter = {
  ALL: `All genres`,
  COMEDY: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Dramas`,
  HORROR: `Horror`,
  KIDS: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLER: `Thrillers`
};

export const FILM_CARD = {
  INIT_STATE_VIEW: 0,
  VIEW_ITEM: 8
};

export const RatingForFilm = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AVESOME: `Avesome`
};

export const RatingNum = {
  AWESOME: 10,
  VERY_GOOD: 8,
  GOOD: 5,
  NORMAL: 3,
  BAD: 0
};
