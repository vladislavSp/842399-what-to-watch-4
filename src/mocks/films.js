const films = [
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Macbeth`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Aviator`,
    genre: `Dramas`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `We need to talk about Kevin`,
    genre: `Thrillers`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Shutter Island`,
    genre: `Horror`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `No Country for Old Men`,
    genre: `Romance`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Snatch`,
    genre: `Sci-Fi`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  promoTitle: `The Grand Budapest Hotel`,
  promoGenre: `Comedy/Drama`,
  promoYear: 2014,
};

export {films, promoFilm};
