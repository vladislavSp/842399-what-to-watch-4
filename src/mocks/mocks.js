const films = [
  {
    id: 0,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 1,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 3,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 5,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 6,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 7,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  id: `1`,
  title: `The Grand Budapest Hotel`,
  genre: `Comedy/Drama`,
  year: 2014,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`
};

const filmContent = {
  title: `Gangs of new york`,
  producer: `Martin Scorsese`,
  genre: `Crime`,
  year: 2002,
  rating: `8.8`,
  actors: [
    `Leonardo DiCaprio`,
    `Cameron Diaz`,
    `Daniel Day-Lewis`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`
  ],
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const reviews = [
  {
    id: 0,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  },
  {
    id: 1,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  }
];

export const userObj = {
  id: `1`,
  avatarUrl: `avatar`,
  email: `email`,
  name: `name`
};

export {films, promoFilm, filmContent, reviews};
