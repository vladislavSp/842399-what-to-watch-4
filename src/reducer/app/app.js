import {GenreFilter, FILM_CARD} from '../../const/const.js';
import {extend} from '../../utils/utils.js';

const initialState = {
  activeGenre: GenreFilter.ALL,
  filmLength: FILM_CARD.VIEW_ITEM,
};

const ActionType = {
  SET_JENRE: `SET_JENRE`,
  SET_VIEW_FILM_CARD: `SET_VIEW_FILM_CARD`,
};

const ActionCreator = {
  setGenreType: (genreType) => ({
    type: ActionType.SET_JENRE,
    payload: genreType,
  }),

  setViewFilmCard: (num) => ({
    type: ActionType.SET_VIEW_FILM_CARD,
    payload: num,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_VIEW_FILM_CARD:
      return extend(state, {filmLength: action.payload + FILM_CARD.VIEW_ITEM});
    case ActionType.SET_JENRE:
      if (state.activeGenre !== action.payload) {
        return extend(state, {activeGenre: action.payload});
      }
  }
  return state;
};

export {reducer, ActionCreator, ActionType};
