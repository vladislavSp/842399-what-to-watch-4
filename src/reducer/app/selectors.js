import NameSpace from '../name-space';
import {genreFilter, MAX_GENRES_NUM, MAX_FILM_CARD} from '../../const/const';
import {getAllFilms} from '../data/selectors';
import {createSelector} from 'reselect';

const NAME_SPACE = NameSpace.APP;

export const getActiveGenre = (state) => {
  return state[NAME_SPACE].activeGenre;
};

export const getCurrentPage = (state) => {
  return state[NameSpace.APP].currentAppPage;
};

export const getViewFilmCard = (state) => {
  return state[NameSpace.APP].filmLength;
};

export const getFilteredFilms = createSelector(
    getAllFilms,
    getActiveGenre,
    (allFilms, currentGenre) => {
      if (currentGenre === genreFilter.ALL) {
        return allFilms;
      }
      return allFilms.filter((film) => film.genre === currentGenre);
    }
);

export const getCurrentViewFilmCard = createSelector(
    getViewFilmCard,
    getFilteredFilms,
    (lengthState, lengthFilms) => {
      if (lengthFilms > lengthState) {
        return lengthState + MAX_FILM_CARD;
      }
      return lengthState;
    }
);

export const getGenres = createSelector(
    getAllFilms,
    (allFilms) => {
      const uniqueGenre = new Set();
      allFilms.forEach((film) => uniqueGenre.add(film.genre));
      const genres = Array.from(uniqueGenre);
      genres.sort().splice(MAX_GENRES_NUM);
      genres.unshift(genreFilter.ALL);

      return genres;
    }
);
