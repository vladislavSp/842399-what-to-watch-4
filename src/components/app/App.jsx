import React from 'react';
import PropTypes from 'prop-types';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import {connect} from 'react-redux';
import Main from '../main/Main.jsx';
import Movie from '../movie-detail/Movie.jsx';
import SignIn from '../sign/SignIn.jsx';
import {getPromoFilm, getAllFilms} from '../../reducer/data/selectors';
import {getFilteredFilms, getCurrentPage} from '../../reducer/app/selectors';
import {getAuthorizationStatus, getAuthorizationError} from "../../reducer/user/selectors";
import {Operation as UserOperation} from '../../reducer/user/user';
import {ActionCreator as AppActionCreator} from '../../reducer/app/app';
import {APP_PAGE} from '../../const/const';

export const App = (props) => {
  const {promoFilm, filteredFilms, authorizationStatus, authorizationError, currentAppPage, login, onSignInClick} = props;

  const renderMainScreen = () => {
    let appPageRender;

    switch (currentAppPage) {
      case APP_PAGE.MAIN_PAGE:
        appPageRender = (<Main
          onSignInClick = {onSignInClick}
          authorizationStatus={authorizationStatus}
          promoFilm={promoFilm}
          films={filteredFilms}
          onCardClick={() => {}}
        />);
        break;
      case APP_PAGE.MOVIE:
        appPageRender = (<Movie
          film={filteredFilms[0]}
          onCardClick={() => {}}
        />);
        break;
      case APP_PAGE.SIGN_IN:
        appPageRender = (<SignIn
          authorizationError={authorizationError}
          onSubmit={login}
        />);
    }

    return appPageRender;
  };

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          {renderMainScreen()}
        </Route>
        <Route exact path="/movie">
          {<Movie
            film={filteredFilms[0]}
            onCardClick={() => {}}
          />}
        </Route>
        <Route exact path="/login">
          {<SignIn
            authorizationError={authorizationError}
            onSubmit={login}
          />}
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  authorizationError: getAuthorizationError(state),
  currentAppPage: getCurrentPage(state),
  promoFilm: getPromoFilm(state),
  films: getAllFilms(state),
  filteredFilms: getFilteredFilms(state).slice(0, 8),
});

const mapDispatchToProps = (dispatch) => ({
  login(authData) {
    dispatch(UserOperation.login(authData));
  },
  onSignInClick() {
    dispatch(AppActionCreator.changeAppPage(APP_PAGE.SIGN_IN));
  }
});

App.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  authorizationError: PropTypes.bool.isRequired,
  currentAppPage: PropTypes.string.isRequired,
  onSignInClick: PropTypes.func.isRequired,
  login: PropTypes.func.isRequired,
  promoFilm: PropTypes.object.isRequired,
  filteredFilms: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
