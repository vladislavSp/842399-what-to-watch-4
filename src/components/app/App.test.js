import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {App} from './app.jsx';
import {genreFilter} from '../../const/const.js';
import {films} from '../../mocks/mocks';
import NameSpace from '../../reducer/name-space';

const mockStore = configureStore([]);

describe(`App render component correctly`, () => {
  it(`Render App / snapshot`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        allFilms: films,
        promoFilm: films[0],
      },
      [NameSpace.APP]: {
        activeGenre: genreFilter.ALL,
        filmLength: 8
      },
      [NameSpace.USER]: {
        authorizationStatus: `NO_AUTH`,
        authorizationError: false,
        user: {}
      }
    });

    const tree = renderer
      .create((
        <Provider store={store}>
          <App />
        </Provider>), {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
