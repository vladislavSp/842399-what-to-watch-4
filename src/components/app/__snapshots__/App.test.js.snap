// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render component correctly Render App / snapshot 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Gangs of new york poster"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Gangs of new york
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              2002
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              href="/films/0/player"
              onClick={[Function]}
            >
              <button
                className="btn btn--play movie-card__button"
                type="button"
              >
                <svg
                  height="19"
                  viewBox="0 0 19 19"
                  width="19"
                >
                  <use
                    xlinkHref="#play-s"
                  />
                </svg>
                <span>
                  Play
                </span>
              </button>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              autoPlay={false}
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
